e2e/cypress/integration/api/iam/04_projects_api_client_runs_ingestion.ts

```diff
@@ -0,0 +1,291 @@
+        },
+        method: 'GET',
+        url: '/api/v0/cfgmgmt/nodes?pagination.size=10'
+      }).then((response) => {
+        expect(response.body).to.have.length(0);
+      });
+    });
+
+    // Ingest a node with attribues that match all the projects
+    cy.fixture('converge/avengers1.json').then((node) => {
+      node.organization_name = '75th Rangers';
+      node.chef_server_fqdn = 'example.org';
+      node.node.chef_environment = 'arctic';
+      node.policy_group = 'red_ring';
+      node.policy_name = 'fire';
+      node.node.automatic.roles = ['backend'];
+      node.node.normal.tags = ['v3'];
+      node.entity_uuid = nodeId;
+      cy.request({
+        headers: { 'api-token': Cypress.env('ADMIN_TOKEN') },
+        method: 'POST',
+        url: '/data-collector/v0',
+        body: node
+      });
+    });
+
+    // wait for the node to be ingested
+    waitForNodes(projectsWithRule[0].project.id, 30);
+
+    // Ensure the node is tagged with the correct project
+    projectsWithRule.forEach(projectWithRule => {
+      cy.request({
+        headers: {
+          'api-token': Cypress.env('ADMIN_TOKEN'),
+          projects: projectWithRule.project.id
+        },
+        method: 'GET',
+        url: '/api/v0/cfgmgmt/nodes?pagination.size=10'
+      }).then((response) => {
+        expect(response.body).to.have.length(1);
+        expect(response.body[0].id).to.equal(nodeId);
+      });
+    });
+
+    // Delete node
+    cy.request({
+      headers: { 'api-token': Cypress.env('ADMIN_TOKEN') },
+      method: 'POST',
+      url: 'api/v0/ingest/events/chef/node-multiple-deletes',
+      body: {
+        node_ids: [nodeId]
+      },
+      failOnStatusCode: false
+    });
+  });
+});
+
+function waitUntilApplyRulesNotRunning(attempts: number): void {
+  if (attempts === -1) {
+    throw new Error('apply-rules never finished');
+  }
+  cy.request({
+    headers: { 'api-token': Cypress.env('ADMIN_TOKEN') },
+    url: '/apis/iam/v2beta/apply-rules'
+  }).then((response) => {
+    if (response.body.state === 'not_running') {
+      return;
+    } else {
+      cy.log(`${attempts} attempts remaining: waiting for apply-rules to be not_running`);
+      cy.wait(1000);
+      waitUntilApplyRulesNotRunning(--attempts);
+    }
+  });
+}
+
+function waitForNodes(project: string, maxRetries: number) {
+  cy
+    .request({
+      headers: {
+        projects: project,
+        'api-token': Cypress.env('ADMIN_TOKEN')
+      },
+      method: 'GET',
+      url: '/api/v0/cfgmgmt/nodes?pagination.size=10'
+    })
+    .then((resp: Cypress.ObjectLike) => {
+      // to avoid getting stuck in an infinite loop
+      if (maxRetries === 0) {
+        return;
+      }
+      if (resp.body.length > 0 ) {
+        return;
+      }
+      cy.wait(1000);
+      waitForNodes(project, maxRetries - 1);
+    });
+}
+
+function uuidv4() {
+  return 'xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx'.replace(/[xy]/g, function(c) {
+    // tslint:disable 
```

Comment:
what lint is disabled here? i think you can specify rules like this `// tslint:disable-next-line:max-line-length`

Reply:
Lint would not allow bit operations.

Reply:
I will look into adding a specific lint exclusion.

Reply:
You are right this comment works `// tslint:disable-next-line:no-bitwise`. I am adding this function to the commands.ts also. The next PR will have a lot of clean up of all the duplicated functions. Thanks for your help.